src/interface/src/app/plan.service.ts

```diff
@@ -0,0 +1,91 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import {
+  BehaviorSubject,
+  catchError,
+  EMPTY,
+  map,
+  Observable,
+  of,
+  take,
+  tap,
+} from 'rxjs';
+
+import { BasePlan, Plan } from './types';
+
+export interface PlanState {
+  all: {
+    [planId: string]: Plan;
+  };
+  currentPlanId: Plan['id'] | null;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PlanService {
+  // Warning: do not mutate state!
+  readonly planState$ = new BehaviorSubject<PlanState>({
+    all: {}, // All plans indexed by id
+    currentPlanId: null,
+  });
+  tempPlanId = 0;
+
+  constructor() {}
+
+  /** Makes a request to the backend to create a plan and updates state. */
```

Comment:
Will there be separate calls for Create vs. Save? N

Reply:
Yep, will be making more of these for Update and Delete.